scripts/policy/frameworks/telemetry/log.zeek

```diff
@@ -0,0 +1,190 @@
+		name: string &log;
+
+		## The unit of this metric, or unset if unit-less.
+		unit: string &log &optional;
+
+		## The names of the individual labels.
+		labels: vector of string &log;
+
+		## The values of the labels as listed in ``labels``.
+		label_values: vector of string &log;
+
+		## The value of this metric.
+		value: double &log;
+	};
+
+	type HistogramInfo: record {
+		## Timestamp of reporting.
+		ts: time &log;
+
+		## Peer that generated this log.
+		peer: string &log;
+
+		## The prefix (namespace) of the metric.
+		prefix: string &log;
+
+		## The name of the metric.
+		name: string &log;
+
+		## The unit of this metric, or unset if unit-less.
+		unit: string &log &optional;
+
+		## The names of the individual labels.
+		labels: vector of string &log;
+
+		## The values of the labels as listed in ``labels``.
+		label_values: vector of string &log;
+
+		## The bounds of the individual buckets
+		bounds: vector of double &log;
+
+		## The number of observations within each individual bucket.
+		values: vector of double &log;
+
+		## The sum over all observations
+		sum: double &log;
+
+		## The total number of observations.
+		observations: double &log;
+	};
+
+	## A default logging policy hook for the stream.
+	global log_policy: Log::PolicyHook;
+
+	## A default logging policy hook for the histogram stream.
+	global log_policy_histogram: Log::PolicyHook;
+
+	## Event triggered for every record in the stream.
+	global log_telemetry: event(rec: Info);
+
+	## Event triggered for every record in the histogram stream.
+	global log_telemetry_histogram: event(rec: HistogramInfo);
+
+	## How often metrics are reported.
+	option log_interval = 60sec;
+
+	## Only metrics with prefixes in this set will be included in the
+	## telemetry.log and telemetry_histogram.log files by default.
+	## Setting this option to an empty set includes all prefixes.
+	##
+	## For more fine-grained customization, setting this option to an
+	## empty set and implementing the :zeek:see:`Telemetry::log_policy`
+	## and :zeek:see:`Telemetry::log_policy_histogram` hooks to filter
+	## individual records is recommended.
+	option log_prefixes: set[string] = {"process", "zeek"};
+}
+
+function do_log()
+	{
+	local ts = network_time();
+	local metrics = Telemetry::collect_metrics();
+
+	for ( i in metrics )
+		{
+		local m = metrics[i];
+
+		# Histograms don't have single values, skip over them.
+		if ( m$opts$metric_type == DOUBLE_HISTOGRAM || m$opts$metric_type == INT_HISTOGRAM )
+			next;
+
+		if ( |log_prefixes| > 0 && m$opts$prefix !in log_prefixes )
+			next;
+
+		local rec: Info;
+		rec$ts = ts;
+		rec$peer = peer_description;
+		# Convert 'Telemetry::DOUBLE_COUNTER' to 'counter' by
+		# trimming everything before '_'
+		rec$metric_type = to_lower(gsub(cat(m$opts$metric_type), /^[^_]+_/, ""));
+		rec$prefix = m$opts$prefix;
+		rec$name = m$opts$name;
+		if ( m$opts$unit != "1" )
+			rec$unit = m$opts$unit;
+		else
+			delete rec$unit;
```

Comment:
Do we need this? Wouldn't the field be unset by default?

Reply:
> Do we need this? Wouldn't the field be unset by default?

I switched to `Info()` construction rather than `local rec: Info;`. Not needed anymore and less confusing.